File: backend/TirthSeva.API/Models/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TirthSeva.API.Models
{
    public class Payment
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [ForeignKey("Booking")]
        public int BookingId { get; set; }

        [Required]
        [Range(0, 100000)]
        public decimal Amount { get; set; }

        [Required]
        [StringLength(50)]
        public string PaymentMethod { get; set; } = string.Empty; // Mock, Stripe, PayPal, UPI

        [StringLength(200)]
        public string? TransactionId { get; set; }

        [Required]
        [StringLength(20)]
        public string Status { get; set; } = "Pending"; // Pending, Success, Failed, Refunded

        [StringLength(500)]
        public string? FailureReason { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? CompletedAt { get; set; }

        // Navigation properties
        public virtual Booking Booking { get; set; } = null!;
    }
}

File: backend-spring/src/main/java/com/tirthseva/api/entity/Payment.java
package com.tirthseva.api.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    private Integer id;

    @NotNull
    @Column(name = "BookingId", nullable = false, unique = true)
    private Integer bookingId;

    @NotNull
    @DecimalMin("0")
    @DecimalMax("100000")
    @Column(name = "Amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;

    @NotBlank
    @Size(max = 50)
    @Column(name = "PaymentMethod", nullable = false, length = 50)
    @Builder.Default
    private String paymentMethod = ""; // Mock, Stripe, PayPal, UPI

    @Size(max = 200)
    @Column(name = "TransactionId", length = 200)
    private String transactionId;

    @NotBlank
    @Size(max = 20)
    @Column(name = "Status", nullable = false, length = 20)
    @Builder.Default
    private String status = "Pending"; // Pending, Success, Failed, Refunded

    @Size(max = 500)
    @Column(name = "FailureReason", length = 500)
    private String failureReason;

    @Column(name = "CreatedAt", nullable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "CompletedAt")
    private LocalDateTime completedAt;

    // Navigation property
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "BookingId", insertable = false, updatable = false)
    private Booking booking;
}
